### PR TITLE
Endpoint for fetching integration metadata

### PR DESCRIPTION
**Pull request by 0xdev22**
# Proposed Changes

### CODE REVIEW

**Comment by elffjs** [comment]
Might we mark this as a draft, given that it won’t compile?

File: cmd/devices-api/api.go [hunk 1]
```diff
@@ -193,6 +193,7 @@ func startWebAPI(logger zerolog.Logger, settings *config.Settings, pdb db.Store,
 	v1Auth.Post("/user/devices", userDeviceController.RegisterDeviceForUser)
 
 	v1Auth.Get("/integrations", userDeviceController.GetIntegrations)
+	v1Auth.Get("/integration/:tokenID", userDeviceController.GetIntegrationNFTMetadata)
```
Comment (elffjs) [inline cmd/devices-api/api.go:196]: You don’t want to require authentication for this, it’s for public use.

Could we place this next to the other NFT endpoints?

File: internal/controllers/user_integrations_controller.go [hunk 2]
```diff
@@ -220,6 +220,33 @@ func (udc *UserDevicesController) GetIntegrations(c *fiber.Ctx) error {
 	})
 }
 
+// GetIntegrationByTokenID godoc
+// @Description gets an integration using it's tokenID
+// @Tags        integrations
+// @Produce     json
+// @Success     200 {array} ddgrpc.Integration
+// @Security    BearerAuth
```
Comment (elffjs) [inline internal/controllers/user_integrations_controller.go:228]:
```suggestion
```

File: internal/controllers/user_integrations_controller.go [hunk 3]
```diff
@@ -220,6 +220,33 @@ func (udc *UserDevicesController) GetIntegrations(c *fiber.Ctx) error {
 	})
 }
 
+// GetIntegrationByTokenID godoc
+// @Description gets an integration using it's tokenID
+// @Tags        integrations
+// @Produce     json
+// @Success     200 {array} ddgrpc.Integration
```
Comment (elffjs) [inline internal/controllers/user_integrations_controller.go:227]: Is this the type we’re returning?

File: internal/controllers/user_integrations_controller.go [hunk 4]
```diff
@@ -220,6 +220,33 @@ func (udc *UserDevicesController) GetIntegrations(c *fiber.Ctx) error {
 	})
 }
 
+// GetIntegrationByTokenID godoc
+// @Description gets an integration using it's tokenID
```
Comment (elffjs) [inline internal/controllers/user_integrations_controller.go:224]:
```suggestion
// @Description gets an integration using its tokenID
```

File: internal/controllers/user_integrations_controller.go [hunk 5]
```diff
@@ -220,6 +220,33 @@ func (udc *UserDevicesController) GetIntegrations(c *fiber.Ctx) error {
...
+// @Produce     json
+// @Success     200 {array} ddgrpc.Integration
+// @Security    BearerAuth
+// @Router      /integration/:tokenID [get]
+func (udc *UserDevicesController) GetIntegrationNFTMetadata(c *fiber.Ctx) error {
+	tokenID := c.Params("tokenID")
+
+	uTokenID, err := strconv.ParseUint(tokenID, 10, 64)
+	if err != nil {
+		return fiber.NewError(fiber.StatusInternalServerError, "failed to get integration")
```
Comment (elffjs) [inline internal/controllers/user_integrations_controller.go:235]: I think in this case we can say that the token id isn’t a valid number. Probably not an internal error. (If this happens elsewhere in this file then I am sorry.)

**Review by elffjs — CHANGES_REQUESTED** [review]

**Review by elffjs — CHANGES_REQUESTED** [review]
A few comments.

**Comment by elffjs** [comment]
Oh, remember to regen docs.

File: internal/controllers/user_integrations_controller.go (hunk 5 above)
Comment (elffjs) [reply internal/controllers/user_integrations_controller.go:234]: Was this resolved?

File: internal/controllers/user_integrations_controller.go (hunk 4 above)
Comment (elffjs) [reply internal/controllers/user_integrations_controller.go:224]: Going to unresolve and commit. I'm not sure what happened here.

File: internal/controllers/user_integrations_controller.go (hunk 5 above)
Comment (0xdev22) [reply internal/controllers/user_integrations_controller.go:234]: I moved the function to the `nftController`. Its been resolved there

**Review by elffjs — APPROVED** [review]

**Pull request merged**